### ISSUE TITLE
Feature Request:  add inline checks inside journals to be sent to chat for pf2 and 5e.

### ISSUE THREAD

**Issue by ThingsCouldGetDicey** [League-of-Foundry-Developers/foundryvtt-forien-quest-log#39]
so the issue i have is that i was planning to use the quest logs to track influence but when i copy the inline checks to send to chat from the main ap journal it doesnt get respected within the mod. as you can see here. 

![image](https://github.com/League-of-Foundry-Developers/foundryvtt-forien-quest-log/assets/62528192/84d47e94-296d-47d0-af91-c2063e816f02)
on the left is the journal from the ap and you can see a button that sends to chat the request. but in the journal its missing as i dont think the mod accommodates for it yet.

could this be added?

**Comment by typhonrt** [comment]
Hi @ThingsCouldGetDicey. If I had to take an offhand gander are you using a module for this somehow? The "text enrichment" is functioning with the links created, but the post to chat thing could somehow be a module that specifically targets the journal. If it was based on the "text enrichment" feature of Foundry then it should also work in FQL. So, do respond with more information on how that feature is represented in a journal / actual text that you put in a normal journal / example of the text you are entering into a normal journal and copying to FQL. If you are happening to use another module that adds this feature then it is not so easy to support in FQL.

**Comment by ThingsCouldGetDicey** [comment]
> Hi @ThingsCouldGetDicey. If I had to take an offhand gander are you using a module for this somehow? The "text enrichment" is functioning with the links created, but the post to chat thing could somehow be a module that specifically targets the journal. If it was based on the "text enrichment" feature of Foundry then it should also work in FQL. So, do respond with more information on how that feature is represented in a journal / actual text that you put in a normal journal and copying to FQL. If you are happening to use another module that adds this feature then it is not so easy to support in FQL.

i turned off the mod in question and its still an issue. here is what it looks like. with it saved and the code inside it. 
![image](https://github.com/League-of-Foundry-Developers/foundryvtt-forien-quest-log/assets/62528192/feae0fde-40a1-4d2a-8f51-62f29d4dfe6a)
![image](https://github.com/League-of-Foundry-Developers/foundryvtt-forien-quest-log/assets/62528192/268c2ca7-5e0a-487d-b3c8-578cae9dbe30)

**Comment by ThingsCouldGetDicey** [comment]
its the traits:action:influence or any variation of traits:action: thats not programmed. as it would look like a send to chat button.

**Comment by typhonrt** [comment]
Because it is another module there may be limits to what can be done. It does look like "text enrichment" / link replacement is being used to create the links. Sight unseen if I had to guess the other module might have CSS / styles that target the journal CSS selectors to show the chat and also possibly hook up any JS code that activates when the "to the chat" icon is pressed. 

I can't add code / special handling to FQL to make another 3rd party module work. The other module has to be coded in a way that it works in a more neutral manner. IE quite likely the work / responsibility is on the other module to handle working in FQL which is a text editor area outside of the normal journal area of Foundry.

Do list the name of the other module? Basically it's going to cost me at least an hour to debug another modules interaction and potentially suggest a solution to other module author, but if they are not willing to address the issue then there is not much that I can do on the FQL side.

**Comment by GuitarGuyNick** [comment]
> Because it is another module there may be limits to what can be done. It does look like "text enrichment" / link replacement is being used to create the links. Sight unseen if I had to guess the other module might have CSS / styles that target the journal CSS selectors to show the chat and also possibly hook up any JS code that activates when the "to the chat" icon is pressed.
> 
> I can't add code / special handling to FQL to make another 3rd party module work. The other module has to be coded in a way that it works in a more neutral manner. IE quite likely the work / responsibility is on the other module to handle working in FQL which is a text editor area outside of the normal journal area of Foundry.
> 
> Do list the name of the other module? Basically it's going to cost me at least an hour to debug another modules interaction and potentially suggest a solution to other module author, but if they are not willing to address the issue then there is not much that I can do on the FQL side.

the text that is being copied is from the kingmaker premium ap journal. if that helps.

**Comment by typhonrt** [comment]
Yeah... That's going to be a tough one to get working with FQL, because it's not from a outside 3rd party developer that likely will be more open for correcting / improving integration. IE it's mediated through the core Foundry team and I just don't think they'll take the request seriously. I almost guarantee if you posted an enhancement issue [here](https://github.com/foundryvtt/foundryvtt-premium-content/issues) that it will languish and no interest will be shown in fixing this issue. I suspect if you get any response it will be "sorry won't fix".

I can't even install the module without purchasing it, so I can't easily debug it / examine if there is a potential solution. Even if I spent significant time to find a solution which I should mention again is _not_ my responsibility especially because it is a paid / premium product there is no guarantee that they'd implement it.

So I think the best current course of action is to post an enhancement issue on the Kingmaker / Foundry VTT premium issues repo and see what type of response you get. If they are open to investigating and finding a solution I'm somewhat open to trying to facilitate that effort for anything I can do on my side, but I just don't think it's going to get to this point.

**Comment by GuitarGuyNick** [comment]
> Yeah... That's going to be a tough one to get working with FQL, because it's not from a outside 3rd party developer that likely will be more open for correcting / improving integration. IE it's mediated through the core Foundry team and I just don't think they'll take the request seriously. I almost guarantee if you posted an enhancement issue [here](https://github.com/foundryvtt/foundryvtt-premium-content/issues) that it will languish and no interest will be shown in fixing this issue. I suspect if you get any response it will be "sorry won't fix".
> 
> I can't even install the module without purchasing it, so I can't easily debug it / examine if there is a potential solution. Even if I spent significant time to find a solution which I should mention again is _not_ my responsibility especially because it is a paid / premium product there is no guarantee that they'd implement it.
> 
> So I think the best current course of action is to post an enhancement issue on the Kingmaker / Foundry VTT premium issues repo and see what type of response you get. If they are open to investigating and finding a solution I'm somewhat open to trying to facilitate that effort for anything I can do on my side, but I just don't think it's going to get to this point.

well they are only posting the inline check code in the pf2 system. the pf2 system is unique in that it has inline checks built into it. then 5e copied its idea. its also not just the kingmaker css. i copied from other premium journals like the blood lords adventure. and its the same issue. its just that theres nothing for fql to recognise this code imho.

**Comment by typhonrt** [comment]
Yeah.. There is likely custom code in those modules that target specifically the normal journal area of Foundry for CSS / styles _and_ any programmatic hook up like the "to the chat" button to post a chat message. IE it's only coded to work with the Foundry journal system.

It's not the responsibility of FQL to add custom code supporting other modules. There _might_ be a technical solution, but I just don't think the Foundry core team / folks behind these premium modules will take the effort to improve this such that it also works with FQL even if I put in the effort to help / assist where possible.

**Comment by typhonrt** [comment]
An interim solution while not as nice is just to create a private / GM journal entry / page that can be linked from the GM notes such that you can open up a normal journal page from a link from the GM Notes section of FQL. A bit of an indirect step, but that should work just as well for connecting the premium module specific features. At least that works for stuff that is GM notes related. Anything that would be in the normal quest description that has additional features from a premium module for all to see may or may not benefit from the same technique where applicable.

**Comment by typhonrt** [comment]
@GuitarGuyNick et al,

Upon investigation this is a feature of the pf2e game system that is not possible to implement in FQL presently. If you happen to browse the pf2e repo the functionality is here: [src/scripts/ui/inline-roll-links.ts](https://github.com/foundryvtt/pf2e/blob/master/src/scripts/ui/inline-roll-links.ts). This code is not globally accessible and is used 
directly in various sheets of the game system explicitly for each area of the game system UI where it is enabled. 

You can see where it is used in response to rendering a journal page via hook here: [src/scripts/hooks/render-journal-text-page-sheet.ts](https://github.com/foundryvtt/pf2e/blob/master/src/scripts/hooks/render-journal-text-page-sheet.ts)

If `InlineRollLinks` was exposed globally there would be a small chance to implement this as an enhancement to FQL. Potentially, the pf2e maintainers can be contacted to make this accessible. It's not really my MO to track down custom features implemented in any game system and enable them in FQL if it was even possible. FQL is game system neutral.
